### PR TITLE
Make smaller improvments

### PR DESCRIPTION
**Pull request by broken1arrow**
I made the code easier to read and also facilitated testing different parts by breaking it down into smaller methods. This is just the first of many planned changes, but I won't make too many alterations until I hear your viewpoint on this.

### CODE REVIEW

File: src/main/java/com/tchristofferson/configupdater/ConfigUpdater.java [hunk 1]
```diff
@@ -48,24 +48,15 @@ private static void write(FileConfiguration defaultConfig, FileConfiguration cur
...
-                        continue keyLoop;
-                    }
-                }
+       for (String fullKey : defaultConfig.getKeys(true)) {
+            String indents = KeyUtils.getIndents(fullKey, SEPARATOR);
 
-                writeCommentIfExists(comments, writer, fullKey, indents);
-            }
 
+         if (!ignoredSectionsValues.isEmpty()){
```
Comment (tchristofferson) [inline src/main/java/com/tchristofferson/configupdater/ConfigUpdater.java:55]: Please fix the indenting in the write method.

File: src/main/java/com/tchristofferson/configupdater/ConfigUpdater.java [hunk 2]
```diff
@@ -347,4 +297,61 @@ private static Object getKeyAsObject(String key, Map<Object, Object> sectionCont
...
+     *
+     * @param ignoredSectionsValues The map containing the ignored section-value mappings.
+     * @param bufferedWriter        The writer to write the value to.
+     * @param fullKey               The full key to search for in the ignoredSectionsValues map.
+     * @throws IOException If an I/O error occurs while writing the value.
+     */
+    private static boolean writeIgnoredSectionValueIfExists(final Map<String, String> ignoredSectionsValues, final BufferedWriter bufferedWriter, final String fullKey) throws IOException {
+        for (final Map.Entry<String, String> entry : ignoredSectionsValues.entrySet()) {
+            if (entry.getKey().equals(fullKey)) {
+                bufferedWriter.write(ignoredSectionsValues.get(fullKey) + "\n");
```
Comment (tchristofferson) [inline src/main/java/com/tchristofferson/configupdater/ConfigUpdater.java:331]: Enhance this by moving the if logic before the loop. Loop is only necessary if the fullKey is a sub key, so the else if logic can stay in the loop.

File: src/main/java/com/tchristofferson/configupdater/ConfigUpdater.java [hunk 3]
```diff
@@ -347,4 +297,61 @@ private static Object getKeyAsObject(String key, Map<Object, Object> sectionCont
...
+	 *
+	 * @param bufferedWriter The writer to write the configuration section to.
+	 * @param indents        The string representation of the indentation level.
+	 * @param trailingKey    The trailing key for the configuration section.
+	 * @param currentValue   The current value of the configuration section.
+	 * @throws IOException If an I/O error occurs while writing the configuration section.
+	 */
+	private static void writeConfigurationSection(final BufferedWriter bufferedWriter, final String indents, final String trailingKey, final Object currentValue) throws IOException {
+		bufferedWriter.write(indents + trailingKey + ":");
+		if (!((ConfigurationSection) currentValue).getKeys(false).isEmpty()) {
```
Comment (tchristofferson) [inline src/main/java/com/tchristofferson/configupdater/ConfigUpdater.java:351]: Change the currentValue parameter type to ConfigurationSection, and cast it where this method is called.

**Review by tchristofferson — CHANGES_REQUESTED** [review]

**Comment by tchristofferson** [comment]
I requested some minor changes.

Does this fix any existing bugs? If it does could you please make a reference to them here?

**Comment by broken1arrow** [comment]
> I requested some minor changes.
> 
> Does this fix any existing bugs? If it does could you please make a reference to them here?

This is more fix so it is more easy to read the code, so you don't have loops that are 100 lines long.

I'm also looking on the bug, when key is null. But that is more complicated, because #config.getKey() ignore sections some don't have values set. 

The only fix right now, is to have set something in the original config, it's not a problem if the key on the disk don't have value set.

File: src/main/java/com/tchristofferson/configupdater/ConfigUpdater.java (hunk 2 above)
Comment (broken1arrow) [reply src/main/java/com/tchristofferson/configupdater/ConfigUpdater.java:331]: Alright can move around the logic, I only keep /preserve your logic. But I see that could be improved.

**Comment by broken1arrow** [comment]
Could perhaps improve the loop for the ignored values even more, will check that later on.

**Pull request merged**